service/history/handler.go

```diff
@@ -92,8 +137,8 @@ var (
 func NewHandler(
 	resource resource.Resource,
 	config *config.Config,
-) *Handler {
-	handler := &Handler{
+) *HandlerImpl {
```

Comment:
could we return interface here?

Reply:
As per Go guidelines we should return concrete implementations instead of interfaces.

Reply:
Thanks. Good to know about this